openstack/clustering/v1/clusters/requests.go

```diff
@@ -546,3 +546,25 @@ func AddNodes(client *gophercloud.ServiceClient, id string, opts AddNodesOpts) (
 	r.Header = result.Header
 	return
 }
+
+func (opts RemoveNodesOpts) ToClusterNodeMap(nodeAction string) (map[string]interface{}, error) {
+	return gophercloud.BuildRequestBody(opts, nodeAction)
+}
```

Comment:
Comparing this with `AddNodeOpts`, I see what you're doing here.

If `AddNodeOpts` and `RemoveNodeOpts` take a `nodeAction` parameter, then what actually distinguishes them from one another?

Wouldn't these be the same:

```
removeOpts.ToClusterNodeMap("del_nodes")
addOpts.ToClusterNodeMap("del_nodes")
```

I can think of two options here:

1. Rename `ToClusterNodeMap` to `ToClusterAddNodeMap` and `ToClusterRemoveNodeMap` and hard-code the `add_nodes` and `del_node` string.

    The benefit of this is creating distinct add and remove interfaces which will never clash.

2. Rename `AddNodeOpts` to just `NodeOpts`, keep `ToClusterNodeMap` as-is. Get rid of `RemoveNodeOpts`.

    The benefit is that you are sharing a common struct between actions that take a list/array of nodes. If there are further actions which take a list/array of nodes, this might be the better way to go.

Thoughts?

Reply:
@jtopjian good points. I prefer to use the second option with code sharing, but might ended up with option 1 because if these api diverges, then will have to use option 1 anyway, so might as well start with option 1.